pipeline-runner/tests/testthat/test-qc-6-integrate_scdata.R

```diff
@@ -170,17 +166,92 @@ test_that("remove_genes removes the correct genes when there are genes to remove
 
   expect_equal(nrow(res), nrow(scdata) - 17)
   expect_false(any(c(some_cc_genes, exclude_custom) %in% rownames(res)))
-
 })
 
 
 test_that("remove_genes doesn't modify the object when there are no matches", {
-
   scdata <- suppressWarnings(mock_scdata())
 
   # empty integer vector
   res <- remove_genes(scdata, exclude_groups = "cellCycle")
 
   expect_equal(res, scdata)
+})
+
+
+test_that("SeuratV4 integration works", {
+  # mock a bigger dataset to run Seurat v4 integration without skipping it
+  scdata <- mock_scdata(n_rep = 3)
+
+  npcs <- get_npcs(scdata)
+
+  config <- list(
+    dimensionalityReduction = list(numPCs = npcs, method = "rpca"),
+    dataIntegration = list(method = "seuratv4", methodSettings = list(seuratv4 = list(numGenes = 1000, normalisation = "logNormalize")))
+  )
+
+  scdata <- suppressWarnings(run_dataIntegration(scdata, config))
+  expect_s4_class(scdata, "Seurat")
+})
+
+
+test_that("PCA is computed when RPCA method is selected within SeuratV4 integration", {
+  # mock a bigger dataset to run Seurat v4 integration without skipping it
+  scdata <- mock_scdata(n_rep = 3)
+
+  npcs <- get_npcs(scdata)
+
+  config <- list(
+    dimensionalityReduction = list(numPCs = npcs, method = "rpca"),
+    dataIntegration = list(method = "seuratv4", methodSettings = list(seuratv4 = list(numGenes = 1000, normalisation = "logNormalize")))
+  )
+
+  expect_message(run_dataIntegration(scdata, config),"Running PCA")
+})
+
+
+test_that("PCA is not computed when CCA method is selected within SeuratV4 integration", {
+  # mock a bigger dataset to run Seurat v4 integration without skipping it
+  scdata <- mock_scdata(n_rep = 3)
+
+  npcs <- get_npcs(scdata)
+
+  config <- list(
+    dimensionalityReduction = list(numPCs = npcs, method = "cca"),
+    dataIntegration = list(method = "seuratv4", methodSettings = list(seuratv4 = list(numGenes = 1000, normalisation = "logNormalize")))
+  )
+
+  expect_message(run_dataIntegration(scdata, config), "PCA is not running .*")
+})
```

Comment:
I'm not a fan of testing using messages printed by us... isn't there a better way? testing the object directly?

Reply:
hmmm can't think of a simple way though. maybe getting into counting calls to a function, but I do not think it's worth it getting into that

Reply:
> I'm not a fan of testing using messages printed by us... isn't there a better way? testing the object directly?

yeah I thought about it, but the problem is we can't test the existence of the PCA slot because it will exist anyway, since PCA is run after the integration with both CCA and RPCA